app/models/version.rb

```diff
@@ -81,8 +81,8 @@ def estimated_hours
   def spent_hours
     @spent_hours ||= TimeEntry
       .not_ongoing
-      .includes(:work_package)
-      .where(work_packages: { version_id: id })
+      .joins("INNER JOIN work_packages ON entity_type = 'WorkPackage' AND work_packages.id = entity_id")
+      .where(work_packages: { version_id: id }, entity_type: "WorkPackage")
```

Comment:
You shouldn't need the additional filtering in `.where` since you're already joining on `entity_type = 'WorkPackage`, Additionally, table aliasing might things a bit clearer.

```suggestion
      .joins("INNER JOIN work_packages ON work_packages.id = time_entries.entity_id AND time_entries.entity_type = 'WorkPackage'")
      .where(work_packages: { version_id: id })
```

Reply:
Again, an optimization. Yes, as we are inner joining, entries with a type != WorkPackage would be filtered out, but that would only happen after loading all t ime entries. So we can pre filter here.